build/source/engine/allocspace.f90

```diff
@@ -349,18 +351,21 @@ subroutine allocateDat_dp(metadata,nSnow,nSoil,nLayers, & ! input
 
   ! allocate structures
   else
-   select case(trim(metadata(iVar)%vartype))
-    case('scalarv'); allocate(varData%var(iVar)%dat(1),stat=err)
-    case('wLength'); allocate(varData%var(iVar)%dat(nBand),stat=err)
-    case('midSnow'); allocate(varData%var(iVar)%dat(nSnow),stat=err)
-    case('midSoil'); allocate(varData%var(iVar)%dat(nSoil),stat=err)
-    case('midToto'); allocate(varData%var(iVar)%dat(nLayers),stat=err)
-    case('ifcSnow'); allocate(varData%var(iVar)%dat(0:nSnow),stat=err)
-    case('ifcSoil'); allocate(varData%var(iVar)%dat(0:nSoil),stat=err)
-    case('ifcToto'); allocate(varData%var(iVar)%dat(0:nLayers),stat=err)
-    case('routing'); allocate(varData%var(iVar)%dat(nTimeDelay),stat=err)
-    case('unknown'); allocate(varData%var(iVar)%dat(0),stat=err)  ! unknown=special (and valid) case that is allocated later (initialize with zero-length vector)
-    case default; err=40; message=trim(message)//"unknownVariableType[name='"//trim(metadata(iVar)%varname)//"'; type='"//trim(metadata(iVar)%vartype)//"']"; return
+   select case(metadata(iVar)%vartype)
+    case(iLookVarType%scalarv); allocate(varData%var(iVar)%dat(1),stat=err)
+    case(iLookVarType%wLength); allocate(varData%var(iVar)%dat(nBand),stat=err)
+    case(iLookVarType%midSnow); allocate(varData%var(iVar)%dat(nSnow),stat=err)
+    case(iLookVarType%midSoil); allocate(varData%var(iVar)%dat(nSoil),stat=err)
+    case(iLookVarType%midToto); allocate(varData%var(iVar)%dat(nLayers),stat=err)
+    case(iLookVarType%ifcSnow); allocate(varData%var(iVar)%dat(0:nSnow),stat=err)
+    case(iLookVarType%ifcSoil); allocate(varData%var(iVar)%dat(0:nSoil),stat=err)
+    case(iLookVarType%ifcToto); allocate(varData%var(iVar)%dat(0:nLayers),stat=err)
+    case(iLookVarType%routing); allocate(varData%var(iVar)%dat(nTimeDelay),stat=err)
+    case(iLookVarType%unknown); allocate(varData%var(iVar)%dat(0),stat=err)  ! unknown=special (and valid) case that is allocated later (initialize with zero-length vector)
+    case default
+     err=40; message=trim(message)//"unknownVariableType[name='"//trim(metadata(iVar)%varname)//"'; type='"//trim(get_varTypeName(metadata(iVar)%vartype))//"']"
+     print*,'!!!!!!!!!!!!!!!!!!!',metadata(iVar)%vartype,metadata(iVar)%varName
```

Comment:
What is the purpose of this print statement? Isn't all information in the error message already?